packages/ui/src/components/collectibles/native/collectible-type-icon-overlay.native.tsx

```diff
@@ -0,0 +1,51 @@
+import { PropsWithChildren, ReactElement } from 'react';
+
+import { NonFungibleCryptoAsset } from '@leather.io/models';
+
+import { OrdinalAvatarIcon } from '../../avatar/ordinal-avatar-icon.native';
+import { StampAvatarIcon } from '../../avatar/stamp-avatar-icon.native';
+import { StxAvatarIcon } from '../../avatar/stx-avatar-icon.native';
+import { Box } from '../../box/box.native';
+
+const overlayOffset = 12;
+
+function getCollectibleTypeIcon(protocol: NonFungibleCryptoAsset['protocol']): ReactElement | null {
+  switch (protocol) {
+    case 'stamp':
+      return <StampAvatarIcon size="sm" />;
+    case 'inscription':
+      return <OrdinalAvatarIcon size="sm" />;
+    case 'sip9':
+      return <StxAvatarIcon size="sm" />;
+    default:
+      return null;
+  }
+}
+
+interface CollectibleTypeIconOverlayProps extends PropsWithChildren {
+  protocol: NonFungibleCryptoAsset['protocol'];
+}
+
+export function CollectibleTypeIconOverlay({
```

Comment:
Is there a reason why this component is in the `ui` and not the app? UI should stay domain-agnostic (this rule is already broken in few places, something we're planning to eliminate). 
Dependency on domain-specific code/packages like `models` is extremely problematic.